File: race-timer-client-web/src/models/SyncStatus.ts
/**
 * Synchronization status for local documents.
 */
export enum SyncStatus {
  /** Document is synchronized with server */
  SYNCED = 'SYNCED',

  /** Document has local changes pending upload */
  PENDING = 'PENDING',

  /** Document is currently being synchronized */
  SYNCING = 'SYNCING',

  /** Synchronization failed (will retry) */
  FAILED = 'FAILED',

  /** Conflict detected between local and server versions */
  CONFLICT = 'CONFLICT'
}

File: race-timer-client/src/main/java/org/rowtown/rms/rrr/client/model/SyncStatus.java
package org.rowtown.rms.rrr.client.model;

/**
 * Synchronization status for local entities.
 */
public enum SyncStatus {
    /**
     * Entity is synchronized with server.
     */
    SYNCED,

    /**
     * Entity has local changes pending upload.
     */
    PENDING,

    /**
     * Entity is currently being synchronized.
     */
    SYNCING,

    /**
     * Synchronization failed, needs retry.
     */
    FAILED,

    /**
     * Conflict detected, needs resolution.
     */
    CONFLICT
}
